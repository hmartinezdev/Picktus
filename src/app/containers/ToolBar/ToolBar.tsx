import Button from '@components/Button';
import colors from '@constants/colors';
import React, { Component } from 'react';
import { IToolBarProps } from './ToolBar.type';

class ToolBar extends Component<IToolBarProps> {
  public render(): React.ReactElement<ToolBar> {
    const { signout } = this.props;

    return (
      <div className="toolbar">
        <Button text="disconnect" onClick={signout} />
        <style jsx>{`
          .toolbar {
            display: flex;
            justify-content: center;
            height: 100%;
            width: 100%;
            background-color: ${colors.primary};
          }
        `}</style>
      </div>
    );
  }
}

export default ToolBar;
