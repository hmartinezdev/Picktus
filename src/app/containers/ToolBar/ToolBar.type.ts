import { ThunkResult } from '@store/reducers/user/user-actions';

export interface IDispatchProps {
  signout: () => any;
}

export type IToolBarProps = IDispatchProps;
