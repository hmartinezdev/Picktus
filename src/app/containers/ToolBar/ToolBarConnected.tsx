import { signout } from '@store/reducers/user/user-actions';
import { connect } from 'react-redux';
import ToolBar from './ToolBar';

export default connect(
  undefined,
  { signout }
)(ToolBar);
