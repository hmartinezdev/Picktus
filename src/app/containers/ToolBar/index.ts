import ToolBarConnected from './ToolBarConnected';

export default ToolBarConnected;
