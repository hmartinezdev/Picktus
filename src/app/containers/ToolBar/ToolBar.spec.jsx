import { shallow } from 'enzyme';
import React from 'react';
import ToolBar from './ToolBar';

let wrapper;

const setup = () => shallow(<ToolBar />);

describe('<ToolBar />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
