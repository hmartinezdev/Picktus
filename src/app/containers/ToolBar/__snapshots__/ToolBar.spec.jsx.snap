// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolBar /> should render properly 1`] = `
<div
  className="toolbar"
>
  <Button
    text="disconnect"
  />
  <style
    jsx={true}
  >
    
          .toolbar {
            display: flex;
            justify-content: center;
            height: 100%;
            width: 100%;
            background-color: #2d3047;
          }
        
  </style>
</div>
`;
